remotes/docker/resolver.go

```diff
@@ -231,10 +233,62 @@ func (r *dockerResolver) Resolve(ctx context.Context, ref string) (string, ocisp
 				return "", ocispec.Descriptor{}, errors.Wrapf(err, "%q in header not a valid digest", dgstHeader)
 			}
 			dgst = dgstHeader
+		} else {
+			log.G(ctx).Debug("no Docker-Content-Digest header, fetching manifest instead")
+
+			req, err := http.NewRequest(http.MethodGet, u, nil)
+			if err != nil {
+				return "", ocispec.Descriptor{}, err
+			}
+			req.Header = r.headers
+
+			resp, err = fetcher.doRequestWithRetries(ctx, req, nil)
+			if err != nil {
+				return "", ocispec.Descriptor{}, err
+			}
+			defer resp.Body.Close()
 		}
 
+		// Default to v1 manifest format
```

Comment:
V1 manifest should not be the default, rather check for the specific anomalous conditions and add a comment why that condition is there. The normal format would be stating the registry which this block is correcting for and the date at which it was last tested to still have that issue. So in this case, just check for the `text/plain`.

Also this block can be moved higher so that the digest calculation can be done inside the other block you added, rather than splitting the logic across 2 blocks.

Reply:
Alright made that change.

As for the two block thing I was originally just trying to avoid duplicating the media type parsing code. I've just moved it to a separate function now.